code/modules/mob/living/carbon/inventory.dm

```diff
@@ -155,3 +155,49 @@
 /mob/living/carbon/proc/get_holding_bodypart_of_item(obj/item/I)
 	var/index = get_held_index_of_item(I)
 	return index && hand_bodyparts[index]
+
+/**
+  * Proc called when giving an item to another player
+  *
+  * This handles creating an alert and adding an overlay to it
+  */
+/mob/living/carbon/proc/give()
+	var/obj/item/receiving = get_active_held_item()
+	if(!receiving)
+		to_chat(src, "<span class='warning'>You're not holding anything to give!</span>")
+		return
+	visible_message("<span class='notice'>[src] is offering [receiving]</span>", \
+					"<span class='notice'>You offer [receiving]</span>", null, 2)
+	for(var/mob/living/carbon/C in orange(1, src))
+		if(!CanReach(C))
+			return

```

Comment:
Since anyone can snatch it, use `for(var/mob/living/carbon/C in viewers())`

Try using more descriptive vars than single upper-case letters, though.